aviatrix/resource_aviatrix_transit_gateway.go

```diff
@@ -1727,59 +1756,40 @@ func resourceAviatrixTransitGatewayUpdate(d *schema.ResourceData, meta interface
 		}
 	}
 
-	if gateway.CloudType == goaviatrix.AWS || gateway.CloudType == goaviatrix.AWSGOV || gateway.CloudType == goaviatrix.AZURE {
-		if d.HasChange("tag_list") {
-			tags := &goaviatrix.Tags{
-				ResourceType: "gw",
-				ResourceName: d.Get("gw_name").(string),
-				CloudType:    gateway.CloudType,
-			}
+	if (d.HasChange("tag_list") || d.HasChange("tags")) && !intInSlice(gateway.CloudType, []int{goaviatrix.AWS, goaviatrix.AWSGOV, goaviatrix.AZURE}) {
```

Comment:
Could we refactor this Update logic to all live under a conditional, I think it will be much easier to understand 
```
if d.HasChange("tag_list") || d.HasChange("tags") {
  // do validation
  // get user provided tags
  // call the update API
}
```